tests/testthat/test-variable-catalog.R

```diff
@@ -101,6 +100,96 @@ with_mock_crunch({
                 type=c("numeric", "categorical", "categorical", "multiple_response")
             )))
     })
+
+    test_that("Variable Catagory as.data.frame method", {
+        expect_equal(as.data.frame(varcat[1:3]),
+            data.frame(
+                alias = c("birthyr", "gender", "location"),
+                name  = c("Birth Year", "Gender", "Categorical Location"),
+                type  = c("numeric", "categorical", "categorical"),
+                stringsAsFactors = FALSE
+            ))
+    })
+    test_that("as.data.frame method returns all fields when keys = 'all'", {
+        expect_identical(
+            names(as.data.frame(varcat, keys = "all")),
+            c("name", "discarded", "alias", "type", "id", "description",
+                "notes", "subvariables", "subvariables_catalog", "resolution",
+                "rollup_resolution")
+        )
+    })
+    test_that("As.data.frame method errors correctly", {
+        expect_error(as.data.frame(varcat[1:3], keys = "Not a field at all"),
+            paste(dQuote("Not a field at all"), "is an invalid key for catalogs of class VariableCatalog.")
+        )
+        expect_error(as.data.frame(varcat[1:3], keys = c("banana", "fooey")),
+            paste(
+                serialPaste(dQuote(c("banana", "fooey"))),
+                "are invalid keys for catalogs of class VariableCatalog.")
+        )
+        expect_error(as.data.frame(varcat[1:3], keys = c("name", "fooey")),
+            paste(dQuote("fooey"), "is an invalid key for catalogs of class VariableCatalog.")
+        )
+    })
+
+    test_that("entryToDF works", {
+        entry <- list(
+            alias = "birthyr",
+            name = "Birth Year",
+            type = "numeric")
+        expected_df <- data.frame(alias = "birthyr",
+            name = "Birth Year",
+            type = "numeric",
+            stringsAsFactors = FALSE)
+        list_entry <- list(
+            name = "mymrset",
+            discarded = FALSE,
+            alias = "mymrset",
+            type = "multiple_response",
+            id = "949d2dc7e7a24e6090cc88bb92e1d2fb",
+            description = "Please select all that apply", notes = "",
+            subvariables = list(
+                "mymrset/subvariables/subvar2/",
+                "mymrset/subvariables/subvar1/",
+                "mymrset/subvariables/subvar3/"
+                ),
+            subvariables_catalog = "mymrset/subvariables/"
+            )
+        expected_list_df <- structure(
+            list(
+                name = "mymrset",
+                discarded = FALSE,
+                alias = "mymrset",
+                type = "multiple_response",
+                id = "949d2dc7e7a24e6090cc88bb92e1d2fb",
+                description = "Please select all that apply",
+                notes = "",
+                subvariables = list(
+                    list(
+                        subvariables = list(
```

Comment:
This doesn't look right: too much nesting. Make a test on `varcat` above that asserts what the "subvariables" column in the data.frame should look like. I think that will clarify what the output should be.